src/supautils.c

```diff
@@ -426,4 +442,34 @@ look_for_reserved_role(Node *utility_stmt, List *roles_list)
 			break;
 	}
 	return NULL;
+}
+
+static bool
+reserved_roles_check_hook(char **newval, void **extra, GucSource source)
+{
+	check_parameter(*newval, "supautils.reserved_roles");
+
+	return true;
+}
+
+static bool
+reserved_memberships_check_hook(char **newval, void **extra, GucSource source)
+{
+	check_parameter(*newval, "supautils.reserved_memberships");
+
+	return true;
+}
+
+static void
+check_parameter(char *val, char *name)
+{
+	List *comma_separated_list;
+	
+	if (val != NULL)
+	{
+		if (!SplitIdentifierString(pstrdup(val), ',', &comma_separated_list))
+			EREPORT_INVALID_PARAMETER(name);
```

Comment:
@shinyaaa Thanks! This guc validation is really nice!

One question, this logic here is looking a lot similar to the one we use on the main `ProcessUtility_hook`:

https://github.com/supabase/supautils/blob/32ae743d4a799477b8b45d5536ef956cfcf7849b/src/supautils.c#L128-L130

I was wondering, is it possible to somehow keep it DRY and make the check hook also assign the value?